ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKSourceFactoryTest.java

```diff
@@ -180,6 +181,7 @@ public void shouldBuildNonWindowedTable() {
     when(dataSource.getDataSourceType()).thenReturn(DataSourceType.KTABLE);
     when(keyFormat.isWindowed()).thenReturn(false);
     when(keyFormat.getWindowInfo()).thenReturn(Optional.empty());
+    when(keyFormat.getFormatInfo().getFormat()).thenReturn("asdf");
```

Comment:
nit: make this example more realistic, e.g.,
```suggestion
    when(keyFormat.getFormatInfo().getFormat()).thenReturn("JSON");
```

Also, it'd be nice to add unit test coverage for the code changes in this PR, i.e., to ensure that `InternalFormats.of()` is properly called by the relevant method. Because of the feature flag, the relevant code path isn't actually being exercised by these unit tests at the moment. You can add a new test that sets the feature flag on the `CONFIG` used in these tests, and then validate that the state store formats are as expected.

Also also, I didn't know that Mockito let you mock chained methods in this way! The internet seems split on whether this is a bad practice or not. This unit test is simple enough that I'm not too worried but could be good to keep in mind.

Reply:
From what I'm reading, since `InternalFormats.of()` is called in a static method (`SchemaKSourceFactory.buildTable()`), I can't use `verify()` to check that specifically without bringing in another framework like [PowerMock](https://www.javadoc.io/doc/org.powermock/powermock-api-mockito/1.6.2/org/powermock/api/mockito/PowerMockito.html#verifyStatic--).

As far as testing the formats, that shouldn't be a problem.

Reply:
I'm not suggesting verifying a mock to check that the method is called. I meant that you can check the value of `result.getSourceStep().getStateStoreFormats()` (or whatever you've renamed the last method to) directly, and compare it to what's expected.